gpytorch/kernels/kernel.py

```diff
@@ -517,11 +588,42 @@ def __call__(
             return res
 
         else:
-            if settings.lazily_evaluate_kernels.on():
-                res = LazyEvaluatedKernelTensor(x1_, x2_, kernel=self, last_dim_is_batch=last_dim_is_batch, **params)
+            if (settings.lazily_evaluate_kernels.on() and self._lazily_evaluate) or last_dim_is_batch:
+                num_outputs_per_input = self.num_outputs_per_input(x1_, x2_)
+                named_parameters = tuple(self.named_parameters())
+
+                if last_dim_is_batch:
+                    x1_ = x1_.transpose(-1, -2).unsqueeze(-1)
+                    x2_ = x2_.transpose(-1, -2).unsqueeze(-1)
+
+                if len(named_parameters):
+                    param_names, params = zip(*named_parameters)
+                    param_batch_shapes = [self.batch_shape] * len(params)
```

Comment:
So being able to use `self.batch_shape` here relies on the fact that all parameters have a fully explicit shape (rather than broadcasting over them)?

Reply:
Yes. I think this is generally true for all kernels in GPyTorch. We can add something to the documentation to be more explicit about this.